rust_src/src/editfns.rs

```diff
@@ -226,16 +223,15 @@ pub fn position_bytes(position: LispNumber) -> Option<EmacsInt> {
 /// from adjoining text, if those properties are sticky.
 #[lisp_fn(min = "2")]
 pub fn insert_byte(byte: EmacsInt, count: Option<EmacsInt>, inherit: bool) {
-    if byte < 0 || byte > 255 {
-        args_out_of_range!(byte, 0, 255)
-    }
+    check_range(byte, 0, 255);
+    let byte = byte as u8;
     let buf = ThreadState::current_buffer_unchecked();
-    let toinsert = if byte >= 128 && buf.multibyte_characters_enabled() {
-        EmacsInt::from(raw_byte_codepoint(byte as c_uchar))
+    let toinsert = if buf.multibyte_characters_enabled() {
```

Comment:
Note that I removed the is_ascii check because `Codepoint::from_raw` already does, making it redundant